src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/VtableIndexStubGenerator.cs

```diff
@@ -89,13 +98,36 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
             context.RegisterConcatenatedSyntaxOutputs(generateManagedToNativeStub.Select((data, ct) => data.Item1), "ManagedToNativeStubs.g.cs");
 
+            IncrementalValuesProvider<IncrementalStubGenerationContext> nativeToManagedStubContexts =
+                generateStubInformation
+                .Where(data => data.VtableIndexData.Direction is MarshalDirection.UnmanagedToManaged or MarshalDirection.Bidirectional);
+
+            IncrementalValuesProvider<(MemberDeclarationSyntax, ImmutableArray<Diagnostic>)> generateNativeToManagedStub = nativeToManagedStubContexts
```

Comment:
The `Select` and `Where` are chained above, is there a reason we don't do that there?

Reply:
Yes, we use the results of this filter in 2 places instead of just 1, so I put it in a local so we could reuse it.